src/Test/Html/Events.elm

```diff
@@ -0,0 +1,149 @@
+module Test.Html.Events
+    exposing
+        ( Event(..)
+        , simulate
+        )
+
+{-|
+
+This module allows you to simulate events on Html nodes, the Msg generated
+by the event is returned so you can test it
+
+@docs Event, simulate
+
+-}
+
+import ElmHtml.InternalTypes exposing (ElmHtml(NodeEntry))
+import Json.Decode exposing (decodeString)
+import Json.Encode exposing (bool, encode, object, string)
+import Native.HtmlAsJson
+import Test.Html.Query as Query
+import Test.Html.Query.Internal as QueryInternal
+
+
+{-| Event constructors to simulate events
+-}
+type Event
```

Comment:
We can't bring functions into Elm via decoders, but we can make a dict like you said. I'd prefer a Dict of `Dict Event Json.Value` with these events held in `elm-html-in-elm`. The idea is to get a better representation in elm-html-in-elm where possible